src/openrct2/core/Compression.cpp

```diff
@@ -138,4 +139,137 @@ namespace OpenRCT2::Compression
         inflateEnd(&strm);
         return true;
     }
+
+    /*
+     * Modified copy of ZSTD_COMPRESSBOUND / ZSTD_compressBound() from zstd 1.5.7, with the argument
+     * type changed from size_t (which may be only 32 bits) to uint64_t, and removes the error handling.
+     */
+    static uint64_t zstdCompressBound(uint64_t length)
```

Comment:
Why do we need this to be 64 bit? Nothing should ever get this far, if it does then we are doing something wrong.

Reply:
Just not a fan of duplicating code, the existing one is perfectly usable until it goes beyond 4 GiB which shouldn't happen.

Reply:
After a chat on Discord, I think its okay, its not absolutely necessary but helps avoid pitfalls.